google/cloud/spanner/internal/session_pool.h

```diff
@@ -55,16 +55,21 @@ namespace internal {
 class SessionPool : public std::enable_shared_from_this<SessionPool> {
  public:
   /**
-   * Create a `SessionPool`.
+   * Construct a `SessionPool`.
    *
-   * The parameters allow the `SessionPool` to make remote calls needed to
-   * manage the pool, and to associate `Session`s with the stubs used to
-   * create them. `stubs` must not be empty.
+   * @warning callers should use the `MakeSessionPool()` factory method rather
```

Comment:
Do you want to make the ctor private and befriend `MakeSessionPool()` so that this is required?

Reply:
I did do that initially but it means I can't use `make_shared`, which I guess isn't a big deal, but I also recall a long discussion with Greg about this in a similar context, where he advocated keeping everything public, so that was enough to make me succumb.